### ISSUE TITLE
back button not working on routing exercises

### ISSUE THREAD

**Issue by mdegani** [rangle/ngCourse2#225]
In the exercises navigating back with the back does change the route, but the content in the ngFor disappears.

**Comment by andrewlo** [comment]
This is a router issue where route component lifecycle hooks aren't called when navigating back/forward.

One workaround is this in root component:

```
constructor(private applicationRef: ApplicationRef, private router: Router) {
    this.router.events.subscribe(() => {
      this.applicationRef.zone.run(() => this.applicationRef.tick());
    });
}
```

Hopefully fixed in a future router release.

**Comment by smithad15** [comment]
The workaround is no longer needed. This has been fixed as of [v3.0.0-rc.1](https://github.com/angular/angular/blob/master/modules/%40angular/router/CHANGELOG.md#300-rc1-2016-08-09)
